packages/vite/src/node/config.ts

```diff
@@ -1700,13 +1705,37 @@ export async function loadConfigFromFile(
   }
 }
 
+async function importConfigFile(resolvedPath: string) {
+  const { module, dependencies } = await inlineImport<{
+    default: UserConfigExport
+  }>(resolvedPath, { configFile: false })
+  return {
+    configExport: module.default,
+    dependencies,
+  }
+}
+
+async function bundleAndLoadConfigFile(resolvedPath: string) {
+  const isESM =
+    typeof process.versions.deno === 'string' || isFilePathESM(resolvedPath)
+
+  const bundled = await bundleConfigFile(resolvedPath, isESM)
+  const userConfig = await loadConfigFromBundledFile(
+    resolvedPath,
+    bundled.code,
+    isESM,
+  )
+
+  return {
+    configExport: userConfig,
+    dependencies: bundled.dependencies,
+  }
+}
+
 async function bundleConfigFile(
   fileName: string,
   isESM: boolean,
 ): Promise<{ code: string; dependencies: string[] }> {
-  const isModuleSyncConditionEnabled = (await import('#module-sync-enabled'))
-    .default
-
```

Comment:
I think we need this code. Probably need to do a similar thing on the `inlineImport` side as well.

Reply:
Ah, it's bad merging